test/addressfield-test.js

```diff
@@ -336,6 +336,37 @@
     expect(17);
   });
 
+  test('select a default country', function() {
+    var $cloneAddress = this.address.clone(),
+        $emptyCountry = $cloneAddress.find('.country'),
+        mockJsonData = {
+          "label": "Country",
+          "options": [
+            {
+              "label": "United States",
+              "iso": "US"
+            },
+            {
+              "label": "Canada",
+              "iso": "CA"
+            },
+            {
+              "label": "Japan",
+              "iso": "JA"
+            }
+          ]
+        };
+
+    $emptyCountry.attr("data-country-selected", "US");
```

Comment:
Ah, I see.  I think the code that you modified is never executed in the test; because the country field on the test fixture already has options populated, it returns immediately before executing the loop.  Were it to continue, you would end up with duplicate options.

I think what you need to do is take the fixture clone and remove all `<option>` elements from the country select list before running the plugin in your test case.

It may also be worth updating the docs too, since the default country data attribute only applies to the scenario where you're using the plugin itself to populate the country list.  (Otherwise, there's already a mechanism to choose a default country, which is to do exactly what you're doing in JS but in plain HTML).